File: Data Structures & Algorithms/validate-parentheses/submission-0.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        for (char t : s)
        {
            if (t == '(') st.push(')');
            else if (t == '[') st.push(']');
            else if (t == '{') st.push('}');
            else
            {
                if (st.size() == 0) return false;
                if (st.top() != t) return false;
                st.pop();
            }
        }

        if (st.size() > 0) return false;
        return true;
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-0.cs
public class Solution {
    public int[] DailyTemperatures(int[] temps) {
        var stack = new Stack<int>();
        var res = new int[temps.Length];
        for(int i = 0; i<temps.Length; i++) {
            while(stack.Count>0 && temps[i]>temps[stack.Peek()]) {
                int k = stack.Pop();
                res[k]=i-k;
            }
            stack.Push(i);
        }
        return res;
    }
}
